src/components/CookieWarning.js

```diff
@@ -13,17 +14,55 @@ import { authStore } from 'src/libs/state'
 export const cookiesAcceptedKey = 'cookiesAccepted'
 
 const CookieWarning = () => {
+  const animTime = 0.3
   const signal = useCancellation()
+  const [showWarning, setShowWarning] = useState(false)
+  const [opacity, setOpacity] = useState(0)
   const { cookiesAccepted } = useStore(authStore)
+  const timeout = useRef()
 
   const acceptCookies = acceptedCookies => {
     authStore.update(_.set('cookiesAccepted', acceptedCookies))
   }
 
+  useEffect(() => {
+    if (cookiesAccepted === undefined) {
+      timeout.current = setTimeout(showComponent, 3000)
+    } else {
+      cookiesAccepted ? hideComponent() : showComponent()
+    }
+  })
+
+  const showComponent = () => {
```

Comment:
Could this be simplified using a Transition or CSSTransition component?

https://reactcommunity.org/react-transition-group/

Reply:
Ayy, yes, definitely. I didn't realize this was already in terra-ui, and didn't want to add new dependencies / packages. Let me try it out :)

Reply:
I just pushed an update that will use this now. The one issue I have with this is that it seems like the exit animation isnt working anymore - the component unmounts before it animates. Any ideas what I'm doing wrong?

Reply:
It looks like opacity should be set to 0 for `exiting` in `transitionStyle`. Currently, opacity is set to 0 only when the `Transition` component reaches the `exited` state, so the CSS transition doesn't have a chance to run before `Transition` unmounts it child component.

Reply:
Thank you! That is totally it :)